backend/logic/users/users.js

```diff
@@ -1,13 +1,17 @@
 const mysql = require('../../sql/connection');
 const format = require('../../validation/format');
+const crypto = require('crypto');
 const hasher = require('../../validation/hash');
 
 exports.insertStudentUser = (username, password, email, id) => {
-	return new Promise(function(res, rej) {
+	return new Promise(function (res, rej) {
+
 		// Connect to database
-		let promise = new Promise(function(resolve, reject) {
-			mysql.getConnection(function(err, conn) {
-				if (err) { reject(err); }
+		let promise = new Promise(function (resolve, reject) {
+			mysql.pool.getConnection(function (err, conn) {
+				if (err) {
+					reject(err);
+				}
 				resolve(conn);
```

Comment:
Great job handling the error, I believe this is the way it is done! Just a quick concern that I'm not 100% sure about, does code execute after reject(err) is called? If so, is there potential for it to execute resolve(conn) in this case?

Reply:
Nope, reject makes the promise "return" an error if i am not mistaken

Reply:
Tyrone is right, I think the code after 'reject' will get called. (In this case it's resolve). However, the call `resolve(conn)` won't do anything because the promise it's already been fulfilled. Once a promise is fulfilled, later `resolve` or `reject` call won't have any effects on it. (I might not be 100% correct, better to check the doc)    So generally it's ok to leave it like this, but put a return is definitely a better practice :)